lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +20,187 @@ const getMethodAuthorization = require('./lib/method/authorization');
 const getMethodIntegration = require('./lib/method/integration');
 const getMethodResponses = require('./lib/method/responses');
 
+function caseInsensitive(str) {
+  return { type: 'string', regexp: new RegExp(`^${str}$`, 'i').toString() };
+}
+
+const contentHandlingSchema = { enum: ['CONVERT_TO_BINARY', 'CONVERT_TO_TEXT'] };
+
+const allowedMethods = ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'HEAD', 'DELETE', 'ANY'];
+const methodPattern = new RegExp(`^(?:\\*|${allowedMethods.join('|')})$`, 'i');
+const methodPathPattern = new RegExp(`^(?:\\*|(${allowedMethods.join('|')}) (\\/\\S*))$`, 'i');
+
+const requestParametersSchema = {
+  type: 'object',
+  additionalProperties: {
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArnString' },
+    {
+      type: 'object',
+      properties: {
+        arn: { $ref: '#/definitions/awsArn' },
+        authorizerId: { $ref: '#/definitions/awsCfInstruction' },
+        claims: { type: 'array', items: { type: 'string' } },
+        identitySource: { type: 'string' },
+        identityValidationExpression: { type: 'string' },
+        managedExternally: { type: 'boolean' },
+        name: { type: 'string' },
+        resultTtlInSeconds: { type: 'integer', minimum: 0, maximum: 3600 },
+        scopes: { type: 'array', items: { type: 'string' } },
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+      },
+      required: [],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const corsSchema = {
+  oneOf: [
+    { type: 'boolean' },
+    {
+      type: 'object',
+      properties: {
+        allowCredentials: { type: 'boolean' },
+        cacheControl: { type: 'string' },
+        headers: { type: 'array', items: { type: 'string' } },
+        maxAge: { type: 'integer', minimum: 1 },
+        methods: { type: 'array', items: { enum: allowedMethods } },
```

Comment:
We also need to be case sensitive here.

(we may deprecate this in future PR, but it's what we currently support)

Reply:
are you sure? from [these lines](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js#L413-L427) it seems to me that `cors.methods` is expected to contain uppercased values.

Reply:
Indeed, I must have been looking at some different `methods` property. Ignore this comment, sorry for confusion